File: backend/LimsAuth.Api/Models/Department.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LimsAuth.Api.Models;

/// <summary>
/// 部门/实验室实体
/// </summary>
[Table("departments")]
public class Department
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [Column("code")]
    [MaxLength(50)]
    public string Code { get; set; } = string.Empty;

    [Required]
    [Column("name")]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// 父部门ID(支持层级结构)
    /// </summary>
    [Column("parent_id")]
    public Guid? ParentId { get; set; }

    /// <summary>
    /// 负责人ID
    /// </summary>
    [Column("manager_id")]
    public Guid? ManagerId { get; set; }

    [Column("description")]
    [MaxLength(500)]
    public string? Description { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // 导航属性
    public virtual Department? Parent { get; set; }
    public virtual ICollection<Department> Children { get; set; } = new List<Department>();
    public virtual User? Manager { get; set; }
    public virtual ICollection<User> Users { get; set; } = new List<User>();
}

File: mobile/lib/models/department.dart
class Department {
  final String id;
  final String code;
  final String name;
  final String? parentId;
  final String? managerId;
  final String? description;
  final bool isActive;
  final String? createdAt;
  final List<Department>? children;
  final dynamic manager;

  Department({
    required this.id,
    required this.code,
    required this.name,
    this.parentId,
    this.managerId,
    this.description,
    this.isActive = true,
    this.createdAt,
    this.children,
    this.manager,
  });

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['id']?.toString() ?? '',
      code: json['code']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      parentId: json['parentId']?.toString(),
      managerId: json['managerId']?.toString(),
      description: json['description']?.toString(),
      isActive: json['isActive'] as bool? ?? true,
      createdAt: json['createdAt']?.toString(),
      children: (json['children'] as List<dynamic>?)
          ?.map((e) => Department.fromJson(e as Map<String, dynamic>))
          .toList(),
      manager: json['manager'],
    );
  }

  Map<String, dynamic> toJson() => {
        'code': code,
        'name': name,
        'parentId': parentId,
        'managerId': managerId,
        'description': description,
      };
}

class CreateDepartmentRequest {
  final String code;
  final String name;
  final String? parentId;
  final String? managerId;
  final String? description;

  CreateDepartmentRequest({
    required this.code,
    required this.name,
    this.parentId,
    this.managerId,
    this.description,
  });

  Map<String, dynamic> toJson() => {
        'code': code,
        'name': name,
        'parentId': parentId,
        'managerId': managerId,
        'description': description,
      };
}

class UpdateDepartmentRequest {
  final String name;
  final String? managerId;
  final String? description;
  final bool isActive;

  UpdateDepartmentRequest({
    required this.name,
    this.managerId,
    this.description,
    this.isActive = true,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'managerId': managerId,
        'description': description,
        'isActive': isActive,
      };
}
